extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/scan/ScanQueryFrameProcessor.java

```diff
@@ -228,7 +229,14 @@ private void populateFrameWriterAndFlushIfNeeded() throws IOException
     createFrameWriterIfNeeded();
 
     while (!cursor.isDone()) {
-      if (!frameWriter.addSelection()) {
+      boolean addedSelection;
```

Comment:
We should also make this change in all the leaf processors. 
So it can be the scan processor and the `GroupByPreShuffleFrameProcessor`. So this change might be required there as well. 
Or you can send the common code to the superclass `BaseLeafFrameProcessor`.

Reply:
Updated the test case and made the change. Instead of the individual processors, I have added the handling in the individual frame writers so that we won't have to add a block in every processor.